### PR TITLE
Grid search cv for RandomForest

### PR DESCRIPTION
**Pull request by svbrodersen**
For this to work on the DecisionTree aswell, we have one of two options:

Either, we make the DecisionTree file entirely into a .py file instead of .pyx file. Otherwise we have to create a wrapper class of the DecisionTree, which resides inside a .py file.

For some reason, when cythonizing a file (which was accidentally happening to the RandomForest.py previously) then the __init__ function makes some modifications unknown to the user. For example if a float is given as input then it might instead choose to store it as a double.

I might see if I can find another solution, which is why this is left as a do not merge.

### CODE REVIEW

File: test.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
+from adaXT.random_forest import RandomForest
```
Comment (NiklasPfister) [inline test.py:1]: This is only temporary correct?

**Review by NiklasPfister — COMMENTED** [review]
Only scrolled over this. Looks fine, will double check once more once the decisiontree is also fixed.

**Comment by svbrodersen** [comment]
@NiklasPfister 

The DecisionTree is now also compatble with GridSearCV. However, there are some things that I think can be improved. Mainly with the setup of DecisionTree and how it wraps the _DecisionTree. Generally I would want the _DecisionTree to only include the computationally heavy code (such as refitting the leaf nodes), but in functions where it just calls the predictor, I see no reason to have that on the DecisionTree.

I also have to look into performance of this new update. Running the tests seems fine, but would like more concrete data.

**Review by NiklasPfister — APPROVED** [review]
This ended up being a pretty big PR, nice work. I guess there will be another PR with the final naming changes from predict to predictor right?

**Pull request merged**